react/src/tests/Emails-test.js

```diff
@@ -0,0 +1,669 @@
+        error: '',
+        message: '',
+        resending: {
+          is_fetching: true,
+          failed: false,
+          error: {},
+          message: ''
+        },
+        confirming: '',
+        emails: [],
+        email: '',
+      }
+    );
+  });
+
+it("Receives a START_RESEND_EMAIL_CODE_SUCCESS action", () => {
+    expect(
+      emailsReducer(
+        mockState,
+        {
+          type: actions.START_RESEND_EMAIL_CODE_SUCCESS,
+            message: 'emails.resend_success'
+        }
+      )
+    ).toEqual(
+      {
+        is_fetching: false,
+        failed: false,
+        error: '',
+        message: '',
+        resending: {
+          is_fetching: false,
+          failed: false,
+          error: {},
+          message: 'emails.resend_success'
+        },
+        confirming: '',
+        emails: [],
+        email: '',
+      }
+    );
+  });
+
+it("Receives a START_RESEND_EMAIL_CODE_FAIL action", () => {
+    expect(
+      emailsReducer(
+        mockState,
+        {
+          type: actions.START_RESEND_EMAIL_CODE_FAIL,
+           payload: {
+              error: {error:"Bad error"},
+         }
+        }
+      )
+    ).toEqual(
+      {
+        is_fetching: false,
+        failed: false,
+        error: '',
+        message: '',
+        resending: {
+          is_fetching: false,
+          failed: true,
+          error: {error:"Bad error"},
+          message: ''
+        },
+        confirming: '',
+        emails: [],
+        email: '',
+      }
+    );
+  });
+
+});
+
+const getState = () => state;
+const state = {
+    emails: {
+        is_fetching: false,
+        failed: false,
+        error: '',
+        message: '',
+        confirming: '',
+        emails: [],
+        email: '',
+        resending: {
+          is_fetching: false,
+          failed: false,
+          error: {},
+          message: ''
+        },
+    },
+    confirming: '',
+    config : {
+        EMAILS_URL: 'test/localhost',
+        email: 'email@localhost.com',
+        confirming: ''
+    }
+};
+
+import { requestEmails, fetchEmails, requestResend, requestResendEmailCode, saveEmail, sendEmail } from 'sagas/Emails';
+import { put, call, select } from "redux-saga/effects";
+
+describe("Async component", () => {
+
+    it("Sagas requestEmails", () => {
+
+       const generator = requestEmails();
+
+       let next = generator.next(actions.getEmails());
+       expect(next.value).toEqual(put(actions.getEmails()));
+
+        next = generator.next(state);
+        next.value.SELECT.selector = function (state) {
+            return state.config;
+        };
+
+        const config = state => state.config.EMAILS_URL;
+        expect(next.value).toEqual(select(state => state.config));
+
+        const emails = generator.next(fetchEmails(config));
+        expect(emails.value.CALL.args[0]).toEqual(fetchEmails(config));
+
+        next = generator.next(emails);
+        expect(next.value).toEqual(put(emails));
+    });
```

Comment:
In general, all tests for redux sagas have similar problems to the above, they need some work.